src/Data/Validation/Primitives.hs

```diff
@@ -135,13 +136,16 @@ attrName -: validator = validateAttr attrName required
   where required (Just subvalue) = run validator subvalue
         required Nothing = Invalid (errMessage "must_be_present")
 
-(-:?) :: Text.Text -> Validator a -> Validator (Maybe a)
-attrName -:? validator = validateAttr attrName optional
-  where optional (Just value) = Just <$> run validator value
-        optional Nothing = Valid Nothing
+optional :: Text.Text -> Validator a -> Validator (Maybe a)
+attrName `optional` validator = validateAttr attrName opt
+  where opt (Just value) = Just <$> run validator value
+        opt Nothing = Valid Nothing
+
+optionalAndNullable :: Text.Text -> Validator a -> Validator (Maybe a)
+attrName `optionalAndNullable` validator = join <$> attrName `optional` (nullable validator)
 
 infixr 5 -:
```

Comment:
Should we go ahead and replace `-:` with `required` while we're at it?

Reply:
I'm fine with that. I hesitated only because, for upstream consumers, that would be a larger change.

Reply:
That's up to you guys to decide. You're the only upstream consumers :wink:

Reply:
I went ahead and did it.  The change won't be anything major, only tedious at best.